_rules/non-visual-reference-alternative-9bd38c.md

```diff
@@ -0,0 +1,405 @@
+
+```html
+<p>Look at the the wide image which is awesome. Now compare it with the narrow image which isn't.</p>
+<img scr="/test-assets/images/awesome_wide.jfif" alt="Wide photo of an awesome landscape." />
+<img scr="/test-assets/images/Non_awesome_narrow.jpg" alt="Narrow photo of a dull landscape." />
+```
+
+#### Passed Example 8
+
+This paragraph hold the visual reference word "triangle" which is included in the heading of the menu; the fact that the described content is one another page of the same website does not restrict this rule.
+
+```html
+<p>
+	On the <a href="/test-assets/SC1.3.3-triangle-menu-with-heading.html">information page</a> you can find more examples
+	within the triangle menu.
+</p>
+```
+
+#### Passed Example 9
+
+This paragraph hold the visual reference word "star" but there is also a heading "examples" that can be referenced; the fact that the identified content is in an iframe does not restrict this rule.
+
+```html
+<p>More examples can be found when you look underneath the star or you can search for the "Examples" heading</p>
+<iframe src="/test-assets/SC1.3.3-star-with-heading.html"></iframe>
+```
+
+#### Passed Example 10
+
+This parapgraph holds the visual reference word "circle" but in this case it is no instruction so it passes.
+
+```html
+<p>
+	This circle is nice.
+	<span style="height: 25px;
+		     width: 25px;
+		     background-color: #bbb;
+		     border-radius: 50%;
+		     display: inline-block;">
+	</span>
+</p>
+```
+
+#### Passed Example 11
+
+This parapgraph holds the visual reference word "circle" but in this case it is no instruction so it passes. Note that this example is not inapplicable because, despite the fact that the text is not visible, it is included in the accessibility tree.
+
+```html
+<p style="position:absolute; top:-9999em">
+	This circle is nice.
+	<span style="height: 25px;
+		     width: 25px;
+		     background-color: #bbb;
+		     border-radius: 50%;
+		     display: inline-block;">
+	</span>
+</p>
+```
+
+#### Passed Example 12
+
+This parapgraph holds the visual reference word "circle" but in this case it is no instruction so it passes. Note that this example is not inapplicable because, despite the fact that the text is not in the accessibility tree, it is visible.
+
+```html
+<p aria-hidden="true">
+	This circle is nice.
+	<span style="height: 25px;
+		     width: 25px;
+		     background-color: #bbb;
+		     border-radius: 50%;
+		     display: inline-block;">
+	</span>
+</p>
+```
+
+#### Passed Example 13
+
+This parapgraph holds the visual reference word "right". The user is told to find the navigation on the right and the navigation is also correctly identified by a `nav` element.
```

Comment:
Technically, the rule says nothing about being in a correct landmark or stuff. I guess the point here is that the accessible name of a `nav` is going to be "Navigation".

Note that this also relates to what I say on Passed Example 7. I am not sure this example actually passes the rule:
* The test target is not on the same page as textual instruction with non-visual characteristic.
* The target is part of instruction about the web content.
* The visual reference word "right" is not visible in the navigation.

I guess this passes the rule because you consider that the word "navigation" is "textual instruction with non-visual characteristic" for a `nav` element?
How would that be stretch to the limits? (trying to figure out sort of a definition for this) Is navigation a "textual instruction with non-visual characteristic" for
* a `div role="navigation"`?
* a list of links (just `ul`, no special role)? (that example would look exactly the same for sighted users if the `nav` tags are removed)

Reply:
> Technically, the rule says nothing about being in a correct landmark or stuff. I guess the point here is that the accessible name of a `nav` is going to be "Navigation".
>
That's my interpretation too. The role would be navigation, and it would be announced as such.

> Note that this also relates to what I say on Passed Example 7. I am not sure this example actually passes the rule:
> 
> * The test target is not on the same page as textual instruction with non-visual characteristic.

Is the test target the link or the page it takes you to?

> * The target is part of instruction about the web content.
> * The visual reference word "right" is not visible in the navigation.
> 
> I guess this passes the rule because you consider that the word "navigation" is "textual instruction with non-visual characteristic" for a `nav` element?

Screen readers that I know of identify `nav` and they speak "navigation" instead of "nav", and then they complement that role announcement with the contents of the `nav`'s accessible name. So Instead of:

> ... the navigation is also correctly identified by a `nav` element.

We could have:

> ... screen readers will announce the navigation because of the `nav` element.

> How would that be stretch to the limits? (trying to figure out sort of a definition for this) Is navigation a "textual instruction with non-visual characteristic" for
> 
> * a `div role="navigation"`?
> * a list of links (just `ul`, no special role)? (that example would look exactly the same for sighted users if the `nav` tags are removed)


But you would not have the "navigation" announcement, as you would if you had role navigation.